nixos/modules/tasks/network-interfaces.nix

```
@@ -521,7 +521,10 @@ in
     };
 
     networking.hostId = mkOption {
-      default = null;
+      default = lib.optionalString (config.networking.machineId != null) lib.substring 0 8 config.networking.machineId;
+      defaultText = literalExpression ''
+        if config.networking.machineId != null then lib.substring 0 8 config.networking.machineId else null
+      '';
       example = "4e98920d";
       type = types.nullOr types.str;
```

```suggestion
      type = types.nullOr (types.strMatching "[[:xdigit:]]{8}");
```

Just noticed: according to `man 3 gethostid` a hostId must be 32 bit / 4 byte / 8 hex digits, so we should set the type more strictly, similarly to machineId

---

It should be resolved now, please check if I understood correctly :)